### PR TITLE
feat: add simple tasks and config files

### PR DESCRIPTION
**Pull request by k4black**
## Description

The new structure for the course repository under consideration.
1. Task detection by `.task.yml` file
2. Extended and unified checker config and deadlines file 
3. **pipeline** for build and testing stage in the yaml config file
4. .folder_name for special folders

## Motivation and Context

The old structure is outdated, a lot of small deprecated things, some issues not solvable with it, so new. 
Please see the main project for the plan.

Related to
* manytask/checker#43
* manytask/checker#45
* manytask/checker#46
* manytask/checker#47
* manytask/checker#48
* manytask/checker#49
* manytask/checker#50
* manytask/checker#51
* manytask/checker#53

## Contribution Checklist

General
- [x] I have read and agreed on the [CONTRIBUTING](../CONTRIBUTING.md) and [CODE OF CONDUCT](../CODE_OF_CONDUCT.md) documents.
- [x] I have run linters, type checks, import sorting and tests locally.
- [ ] I have added comments to code in hard-to-understand areas.
- [x] Pull Request title uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) format: `tag(scope): description`.

Added tests?
- [ ] yes
- [ ] partially, will add more later
- [ ] no, because I need help
- [ ] no, because they aren't needed

Added to documentation?
- [ ] README.md
- [ ] `docs/`
- [ ] `manytask.github.io` page
- [ ] no documentation needed

### CODE REVIEW

File: .checker.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+version: 1
+
+
+# can be overwritten in .task.yml for individual tasks
+structure:
+  # private patterns exclude from export
+  private_patterns: ["**/.*", "**/test_private.py"]
+  # public patterns to export
+  public_patterns: ["tools", "pyproject.toml", ".gitignore", "**/test_public.py", "**/conftest.py"]
```
Comment (kalabukdima) [inline .checker.yml]: It's not clear what happens to files matching both (or none) patterns. Maybe we should use some regex supporting negative lookahead and merge it into a single list?

File: .checker.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  # explicit patterns allow to change by student
+  allow_to_change_patterns: []
+
+
+# default values for all tasks, can be overwritten in .task.yml params:
+params:
```
Comment (kalabukdima) [inline .checker.yml]: `default_params`?

File: .checker.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  coverage: null  # 0..1, null for no measurement
+
+
+# settings for export command, uses .deadlines.yml and `params` and each task params (in .task.yml)
+export:
+  destination: https://gitlab.manytask.org/test/public-test-repo
+  default_branch: main
+  commit_message: "chore(auto): export new tasks"
+  templates: search  # search or create
+  patterns: ["**", "**/.gitignore", ".gitlab-ci-students.yml", ".docker", "**/.lecture"]
```
Comment (kalabukdima) [inline .checker.yml]: How does it differ from `structure.public_patterns`?

File: .checker.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      run_if: ${{ run_typechecking }}
+      fail: after_all  # fast, after_all, never
+      run: "run_script"
+      isolate: true
+      args:
+        origin: "${{ TEMP_DIR }}/${{ TASK_SUB_PATH }}"
+        script: "python -m mypy --config-file=${{ REPO_DIR }}/pyproject.toml ${{ TASK_DIR }}"
+
+    - name: "Collect tests"
+      run_if: ${{ run_testing }}
```
Comment (kalabukdima) [inline .checker.yml:99]: Maybe those should be nested in some object, like `{{ task.run_testing }}`?

File: .checker.yml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      fail: after_all  # fast, after_all, never
+      run: "run_script"
+      isolate: true
+      args:
+        origin: "${{ TEMP_DIR }}/${{ TASK_SUB_PATH }}"
+        script: "python -m pytest --tb=no -qq --collect-only ${{ TASK_DIR }}"
+        timeout: ${{ timeout }}
+
+    - name: "Run tests"
```
Comment (kalabukdima) [inline .checker.yml:108]: For the grading one possible solution would be to assign a `weight` for each step in the pipeline and use the weighted sum of passed steps as the score. That way you can customize what is making up the score but it still doesn't solve the case with multiple grade types.

File: .checker.yml (hunk 4 above)
Comment (k4black) [reply .checker.yml:99]: Seems it is more complicated to do - you need an real objects there e.g. jinja2

**Pull request merged**